src/Magnum/Math/Random.h

```diff
@@ -0,0 +1,131 @@
+#ifndef Magnum_Math_Random_h
+#define Magnum_Math_Random_h
+
+// TO DO Licence things.
+
+#include <random>
+#include <chrono>
+#include "Magnum/Types.h"
+#include "Magnum/Math/Constants.h"
+#include "Magnum/Math/Vector2.h"
+#include "Magnum/Math/Vector3.h"
+#include "Magnum/Math/Quaternion.h"
+#include "Magnum/Math/Functions.h"
+
+namespace Magnum
+{
+namespace Math
+{
+
+namespace Implementation
+{
+static std::seed_seq seeds{{
+    static_cast<std::uintmax_t>(std::random_device{}()),
+    static_cast<std::uintmax_t>(std::chrono::steady_clock::now()
+                                    .time_since_epoch()
+                                    .count()),
+}};
+
+class RandomGenerator
+{
+public:
+    RandomGenerator() = delete;
+
+    template <typename T>
+    static typename std::enable_if<std::is_same<Int, T>::value, T>::type
+    generate(T start = -Magnum::Math::Constants<T>::inf(),
+             T end = Magnum::Math::Constants<T>::inf())
+    {
+        return std::uniform_int_distribution<T>{start, end}(generator());
+    }
+
+    template <typename T>
+    static typename std::enable_if<std::is_same<Float, T>::value, T>::type
+    generate(T start = -Magnum::Math::Constants<T>::inf(),
+             T end = Magnum::Math::Constants<T>::inf())
+    {
+        return std::uniform_real_distribution<T>{start, end}(generator());
+    }
+
+public:
+    static std::mt19937 &generator()
+    {
+        static std::mt19937 g{seeds};
+        return g;
+    }
+};
+} // namespace Implementation
+namespace Random
+{
+template <class T = Float>
+T randomUnsignedScalar() // range [0, 1]
+{
+    return Implementation::RandomGenerator::generate<T>(static_cast<T>(0.0f),
+                                                        static_cast<T>(1.0f));
+}
+template <class T = Float>
+T randomSignedScalar() // range [-1, 1]
+{
+    return Implementation::RandomGenerator::generate(static_cast<T>(-1.0f), static_cast<T>(1.0f));
+}
+
+template <class T = Float>
+Vector2<T> randomUnitVector2()
+{
+    auto a = Implementation::RandomGenerator::generate(0.0f, 2 * Math::Constants<T>::pi());
+    return {std::cos(a), std::sin(a)};
```

Comment:
I guess this is where things become *hard* :sweat_smile: 

I actually have no idea here -- will the distribution be still uniform if sin/cos is used? I guess it will? Ideally this would be without the extra overhead of trig functions, but I don't have any idea if that's doable ... [in this thread on SO](https://stackoverflow.com/questions/6283080/random-unit-vector-in-multi-dimensional-space) they use a Gaussian distribution as an input, but .. ¯\\\_(ツ)_/¯

If you have better references than me, mention them here please :)

Reply:
For this case sin and cos shall not be get affected ? (sin (+ + - - )[50%],  cos (+ - - + )[50%] )
So this shall be as accurate as the generator itself ? 

Can you also comment about what I read [here](https://mathworld.wolfram.com/SpherePointPicking.html) ? It looks accurate according to this.

Reply:
*Yes* that's a great reference, thanks -- a link to it should go in the documentation. I didn't absorb it fully yet, but yeah it sounds like a good proof. 

> getting 3 random numbers and normalizing

this is what you do for quaternions tho .. can the two-/three-dimensional case be extended for those as well?

Reply:
Quaternion is link [here.](    //http://planning.cs.uiuc.edu/node198.html)

Added both to the code :)